pkg/controller/node.go

```
@@ -161,6 +162,14 @@ func (c *Controller) handleAddNode(key string) error {
 			klog.Errorf("failed to alloc random ip addrs for node %v: %v", node.Name, err)
 			return err
 		}
+
+		// Clean up potentially existing logical switch ports to avoid leftover issues from previous failed configurations
+		if err := c.OVNNbClient.DeleteLogicalSwitchPort(portName); err != nil {
+			klog.Errorf("failed to delete logical switch port %s: %v", portName, err) // Log as error
```

```suggestion
			klog.Errorf("failed to delete stale logical switch port %s: %v", portName, err) // Log as error
```

---

We should return the error.